addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/PresenceDetection.java

```diff
@@ -159,13 +159,39 @@ public void setUseArpPing(boolean enable, String arpPingUtilPath) {
         if (!enable || StringUtils.isBlank(arpPingUtilPath)) {
             arpPingState = "Disabled";
             arpPingMethod = null;
-            return;
+            logger.trace("arpPingState: {}", arpPingState);
+            logger.trace("ARP ping is disabled!");
         } else if (destination == null || !(destination instanceof Inet4Address)) {
-            arpPingState = "Destination is IPv4";
+            arpPingState = "Destination is not a valid IPv4 address";
             arpPingMethod = null;
-            return;
+            logger.trace("arpPingState: {}", arpPingState);
+            logger.trace("ARP ping is disabled!");
+        } else {
+            arpPingMethod = networkUtils.determineNativeARPpingMethod(arpPingUtilPath);
+        }
+        // Set arp_state, which gets displayed in the thing configuration of paperui/habmin
+        switch (arpPingMethod) {
+            case UNKNOWN_TOOL: {
+                arpPingState = "Unknown arping tool";
+                logger.trace("Unknown arping tool without output for --help parameter: {}", arpPingUtilPath);
+                break;
+            }
+            case THOMAS_HABERT_ARPING: {
+                arpPingState = "arping tool by Thomas Habets with timeout";
+                logger.trace("arping tool by Thomas Habets detected: {}", arpPingUtilPath);
+                break;
+            }
+            case THOMAS_HABERT_ARPING_WITHOUT_TIMEOUT: {
```

Comment:
Can you make another improvement by fixing the typo?:

`THOMAS_HABERT_ARPING_WITHOUT_TIMEOUT` -> `THOMAS_HABETS_ARPING_WITHOUT_TIMEOUT`